sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowPartitionsExec.scala

```diff
@@ -53,9 +53,12 @@ case class ShowPartitionsExec(
       var i = 0
       while (i < len) {
         val dataType = schema(i).dataType
-        val partValue = row.get(i, dataType)
-        val partValueStr = Cast(Literal(partValue, dataType), StringType, Some(timeZoneId))
-          .eval().toString
+        val partValueStr = if (row.isNullAt(i)) {
+          "null"
+        } else {
+          Cast(Literal(row.get(i, dataType), dataType), StringType, Some(timeZoneId))
+            .eval().toString
```

Comment:
Actually, NPE occurs on `null.toString`. I am thinking of the case when `Cast(Literal(row.get(i, dataType), dataType), StringType, Some(timeZoneId)).eval()` returns `null` for non-null input. Maybe it is better to check the result of `eval()`, so, it should be much more reliable fix.